src/Nethermind/Nethermind.Core/Collections/CollectionExtensions.cs

```diff
@@ -22,5 +26,17 @@ public static void AddRange<T>(this ICollection<T> list, params T[] items)
                 list.Add(items[index]);
             }
         }
+
+        public static void NoResizeClear<TKey, TValue>(this ConcurrentDictionary<TKey, TValue> dict)
+                where TKey : notnull
+        {
+            // We iterate over the keys and remove them one by one because calling Clear() on
+            // the ConcurrentDictionary resets its capacity to 31 and then it has to constantly resize.
+            foreach (TKey key in dict.Keys)
+            {
+                dict.TryRemove(key, out _);
+            }
```

Comment:
Use `ConcurrentDictionaryLock` class to lock the dictionary

Reply:
Done